histogram/src/histogram.rs

```diff
@@ -55,8 +56,8 @@ impl Builder {
     /// largest power of two that is less than or equal to the provided value.
     /// For example, if the minimum resolution is set to 10, the width of the
     /// smallest bucket will be 8.
-    pub fn min_resolution(mut self, width: u64) -> Self {
-        self.m = 64 - width.leading_zeros();
+    pub fn min_resolution(mut self, width: NonZeroU64) -> Self {
```

Comment:
yeah, while this is nice, i think it might be more convenient to just use conditional logic in this function. I don't think we should panic in the builder. If somebody passes a minimum width of 0 I don't think it's problematic to just set min width to 1 (by using self.m = 0)

```
pub fn min_resolution(mut self, width: u64) -> Self {
        if width > 0 {
            self.m = 63 - width.leading_zeros();
        } else {
            self.m = 0;
        }
        self
    }
```